Add special text for sample in clocked systems. (#3099)
modelica/ModelicaSpecification: chapters/synchronous.tex

Such a partition has to be solved with a \emph{solver method} of \cref{solver-methods}.
When \lstinline!previous(x)! is used on a continuous-time state variable \lstinline!x!, then \lstinline!previous(x)! uses the start value of \lstinline!x! as value for the first clock tick.

The use of the operator \lstinline!sample! from \cref{event-related-operators-with-function-syntax} in a clocked partition is problematic.

> Diagnostics are appropriate especially if it is intended to generate sampling faster than the clocks, and otherwise the sampling should ideally be adjusted to the clock ticks.

Reviewer: We should avoid introducing a new strength level of diagnostics.  Can we reformulate using _recommended_?  (Please check suggestion carefully, as it also suggests some other changes.)
```suggestion
A diagnostic is recommended, especially if the operator is intended to generate events faster than the clock ticks, and otherwise the sampling should ideally be adjusted to the clock ticks.
```

\begin{nonnormative}

> Since the goal is that it should be possible to use \emph{every} continuous-time Modelica model in a sampled data control system, it is not forbidden.

Reviewer: I think _the goal_ needs a bit of context, maybe like so (I failed to come up with a nice formulation using _goal_ – that it is a goal is now implicit):
```suggestion
The reason for not disallowing \lstinline!sample! in a clocked partition is to make it possible to include \emph{any} continuous-time Modelica model in a sampled data control system.
```
